app/components/telegram-auth.tsx

```diff
@@ -311,7 +311,12 @@ export default function TelegramAuth() {
     } catch (err: any) {
       console.error('checking password:', err)
       await getSRP()
-      setError(err)
+      
+      if (err.errorMessage === 'PASSWORD_HASH_INVALID') {
```

Comment:
use our `getErrorMessage` utility function

```ts
  const errorMsg = getErrorMessage(error)
  if (errorMsg.includes('PASSWORD_HASH_INVALID')) {
        setError(new Error('Your password was incorrect. Please try again.'))
  } else {
        setError(err)
  }
```

Can you also add this to the `getErrorMessage` function, it it's located at the `errorhandling.ts` file

```ts
 if (typeof err === 'object' && err !== null) {
    const msg = (err as any).message ?? (err as any).errorMessage
    if (typeof msg === 'string') return msg
  }
```